src/containers/TaskRuns/TaskRuns.js

```diff
@@ -49,93 +57,76 @@ export /* istanbul ignore next */ class TaskRunsContainer extends Component {
   }
 
   state = {
-    notification: null,
-    loading: true,
     selectedStepId: null,
     selectedTaskId: null,
-    taskRuns: []
+    loading: true
   };
 
-  async componentDidMount() {
-    try {
-      const { match } = this.props;
-      const { taskName } = match.params;
-      await this.loadTaskRuns(taskName);
-    } catch (error) {
-      let message = error;
-      if (error.response) {
-        message = error.response.status === 404 ? 'Not Found' : 'Error';
-      }
-      this.setState({
-        notification: { kind: 'error', message },
-        loading: false
-      });
-    }
+  componentDidMount() {
+    const { match } = this.props;
+    const { taskName } = match.params;
+    this.props
+      .fetchTask(taskName)
+      .then(this.props.fetchTaskRuns())
+      .then(() => this.setState({ loading: false }));
   }
 
   componentDidUpdate(prevProps) {
     const { match } = this.props;
     const { taskName } = match.params;
     if (taskName !== prevProps.match.params.taskName) {
-      this.loadTaskRuns(taskName);
+      this.props.fetchTask(taskName).then(this.props.fetchTaskRuns());
```

Comment:
`Promise.all`


probably a good idea to extract this to a function to ensure consistency when it's updated later since it's used in both componentDidMount and here